### ISSUE TITLE
NGAS support multiple file extensions with the same mime-type

### ISSUE THREAD

**Issue by smclay** [ICRAR/ngas#42]
It would appear that NGAS only supports a single file extension per mime-type. It would be desirable if NGAS could support multiple files extensions with the same mime-type. For example bin, dat and exe could all be application/octet-stream, htm and html could be text/html.

**Comment by rtobar** [comment]
@smclay I agree with the sentiment. After a quick browse through the code it seems that the one-to-one mapping between mime types and extensions is relied upon in quite a few places though, so my first guess is that it would take some effort to do this change correctly.

The current way to deal with multiple extensions that belong to the same logical "type" is to register them with slightly different mime types, and then register the same plug-in to act on all the different mime types. For example in the [sample config file](https://github.com/ICRAR/ngas/blob/master/cfg/sample_server_config.xml) different mime types are defined for the `fits`, `fits.gz` and `fits.Z` extensions, but then they are all treated by the same archiving and processing plug-ins. I know it's not ideal, but it should suffice for most use cases.

**Comment by smclay** [comment]
I had a feeling it would not be trivial. Thank you for the explanation. Perhaps this one for the future.
